pointmatcher/DataPointsFilters/utils/utils.h

```diff
@@ -41,6 +41,18 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 namespace utils {
 
+template<class T>
+inline constexpr T pow(const T base, const std::size_t exponent)
+{
+    // (parentheses not required in next line)
```

Comment:
This is overcomplicated and the comment superfluous. A good example of (premature optimization](http://wiki.c2.com/?PrematureOptimization). I'd just go for:
```c++
return exponent == 0 ? 1 : base * pow(base, exponent - 1)
```
Consider that every non-empty line of code (including comment lines!) comes at a costs over its entire lifetime as additional load for maintenance and debugging and thus should justify this cost.

Reply:
Same as above, one of the author suggest using [Exponentiation by squaring](https://en.wikipedia.org/wiki/Exponentiation_by_squaring). But I'm not sure of the benefits, a simpler version should do the same job considering that the evaluation is at the compile time !

Reply:
Yes, assuming compile time evaluation it only has a very small effect on the compile time (but for your examples 2, and 3 it might actually be slower with the squaring!) . 

You don't need to care whether this is actually evaluated at compile time or runtime, because it is only about initializing very few global variables (typically 1 or 2 in a given program). 

Again, beware of the big mistake of premature optimization: applied to this case: if you don't have a performance problem with the simple `constexp pow` function, don't go for something more complicated. And trust me, you won't have a performance issue with this one!